### DISCUSSION TITLE
Artisan reload command with event/hooks to restart services after updates

### DISCUSSION THREAD

**Discussion by barryvdh** [laravel/framework#50900] [Ideas]
## Issue
Currently, each 'service' has its own restart command:

 - php artisan reverb:restart
 - php artisan octane:reload
 - php artisan queue:restart
 - php artisan horizon:terminate
 - php artisan pulse:restart

Deploy scrips (either with Laravel Forge, something like Capistrano/Deployer can be somewhat easily updated, but easy to forget. Manual deployments (eg. `git pull`) make it even more easy to forget one of those, leading to weird behavior.

## Proposal
I propose a `php artisan reload` (or `deploy` / `restart`) to reload all those services at once. Each service should be responsible for registering itself to this reload command. This can be either:
 - Add a command to some service (eg `ReloadManager::addCommand('php artisan reverb:restart')`
 - Dispatch an event (`Event::dispatch(new ReloadingLaravel)`; which packages can hook into.

Something similar is done with `php artisan optimize` although this one is not configurable. This makes it easy for deploy scripts to just add `php artisan reload` and not worry about specific services.

**Comment by Braunson** [comment]
Suggestion: Add these to a custom composer script... `composer reload-services`

```json
"scripts": {
    "reload-services": [
        "@php artisan reverb:restart --quiet || true",
        "@php artisan octane:reload --quiet || true",
        "@php artisan queue:restart --quiet || true",
        "@php artisan horizon:terminate --quiet || true",
        "@php artisan pulse:restart --quiet || true",
    ]
}
```

**Reply by pelmered** [reply]
Yes, this helps a bit as you only need to edit in one place. But you still need to manage this and update this every time you install a package that needs a long running process. Leaving the responsibility for each package to register it self would be much nicer.
